flutter/lib/src/event_processor/platform_exception_event_processor.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:sentry/sentry.dart';
+
+/// Add code & message from [PlatformException] to [SentryException]
+class PlatformExceptionEventProcessor implements EventProcessor {
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {Hint? hint}) {
+    final exceptions = <SentryException>[];
+
+    for (SentryException exception in (event.exceptions ?? [])) {
+      final platformException = exception.throwable;
+      if (platformException is PlatformException) {
+        exception = _enrich(exception, platformException);
+      }
+      exceptions.add(exception);
+    }
+
+    return event.copyWith(exceptions: exceptions);
+  }
+
+  SentryException _enrich(
+      SentryException sentryException, PlatformException platformException) {
+    final data = Map<String, dynamic>.from(
+      sentryException.mechanism?.data ?? {},
+    );
+    data['platformException'] = {'code': platformException.code};
+    if (platformException.message != null) {
+      data['platformException']['message'] = platformException.message;
+    }
```

Comment:
If we put it under the `data` field, we need to be sure that this is possible to make it searchable on Sentry/Snuba.
[Most likely yes](https://github.com/getsentry/sentry/blob/abb44c51917828ad67d0654f4192bbb0c3873a96/src/sentry/rules/conditions/event_attribute.py#L120-L131).
Since the `error.handled` search actually checks the `mechanism` metadata.

Reply:
The data shows up on sentry, but i'm not familiar how to search for it.

<img width="1449" alt="Bildschirm­foto 2023-02-28 um 14 14 42" src="https://user-images.githubusercontent.com/3984453/221865478-45ec3150-5d5d-4f08-a76f-5611443b2ae1.png">

Reply:
Apparently this becomes a tag so it should work OOTB.

Reply:
I am checking internally if we can add it under the `meta`, let's not merge this for now.

Reply:
> Quick question, do we know if the code is actually one of ISO C99, POSIX.1-2001 or POSIX.1-2008. or just a hardcoded in the Flutter framework?

It's not. It is a free form string and pretty much every library using throwing PlatformExceptions has it's own set of error codes.

Reply:
Ok checked internally, and either adding something under `meta` (requires Relay changes) or under `data` would be fine, since PHP already has under `data` and this PR adds under `data`, let's keep it as it is.